src/Service/QueueManager.php

```diff
@@ -140,6 +148,19 @@ public function getNextJob(): Promise
                 /** @var Job $job */
                 $job = yield $this->elasticSearch->fetchJob($jobUuid, $this->flowConfig->getTube());
             } catch (\Throwable $exception) {
+                $notFoundCount = self::$beanstalkIdToNotFoundCountMap[$jobBeanstalkId] ?? 1;
+                if ($notFoundCount <= self::JOB_NOT_FOUND_MAX_RETRIES) {
+                    self::$beanstalkIdToNotFoundCountMap[$jobBeanstalkId] = $notFoundCount + 1;
+                    $this->logger->warning(
+                        "Job with UUID {$jobUuid} not found in Elasticsearch. Retrying...",
+                        ['beanstalk_id' => $jobBeanstalkId, 'not_found_count' => $notFoundCount]
+                    );
+                    yield $this->beanstalkClient->release(
+                        $jobBeanstalkId,
+                        self::JOB_NOT_FOUND_RETRY_DELAY_SECONDS ** $notFoundCount
+                    );
+                    return yield $this->getNextJob();
```

Comment:
Non si rischia un loop di chiamate ricorsive facendo così?

Forse sarebbe meglio wrappare tutto in un loop do-while da cui normalmente si esce subito restituendo il job trovato, ma se ci sono errori si continua a girare finché non si riesce a recuperare un job.

Reply:
Solved!